simpleviewinc/keg-cli
# chore(deps): Update dependencies

**Ticket**: [ZEN-614](https://jira.simpleviewtools.com/browse/ZEN-614)

## Goal

* Update the dependencies of keg-base and keg-core docker images
* Update the ENV's to dependency latest version
  * Update `expo-cli`, `sharp-cli`, and `node`
* Add `KEG_COPY_LOCAL` back to keg-core docker image

## Updates
* `containers/base/Dockerfile`
  * Fix issue with installing python
* `containers/base/defaults.env` && `scripts/ci/ci.env` && `scripts/setup/defaults.env` && `src/__mocks__/injected/injectedTest.js`
  * Update expo-cli, sharp-cli, and node version
* `src/tasks/cli/env/sync.js`
  * Minor update to wording
* `containers/tap/Dockerfile`
  * Removed old code no longer needed

## Tests

### Setup
* Switch the the keg-cli with cmd => `keg cli`
* Pull this pr with  => `keg pr 112`
* Install all deps with => `yarn install`

### Test 1
* Run the update script =>  `keg cli update --version 5.2.0`
* Open your global config => `keg config open`
  * Then open your `defaults.env` files
  * Ensure the following ENVs were updated to the correct values
    * EXPO_CLI_VERSION=4.10.0
    * SHARP_CLI_VERSION=1.15.0
    * KEG_NODE_VERSION=14.17-alpine
  * [Looks like this](https://i.imgur.com/VFK7N3r.png)

### Test 2
* Run the command `yarn test`
  * Ensure all tests pass

### Test 3
* Run the command `keg dpg run keg-core:ZEN-614-update-core`
  * This image is using the updated node / expo-cli / sharp-cli version
  * Navigate to [the keg-core app](http://keg-core-zen-614-update-core.local.keghub.io/) in your browser
  * Ensure the container works as expected

### Test 4
* **Notes** - This test is the same as the test in [this PR](https://github.com/simpleviewinc/keg-hub/pull/184)
  * So if you already did that PR's Tests, then your can skip this one
* Run keg-core inside a tap by running `keg evf pack run evf:zen-614-tap-test`
  * It should pull and run the image
  * Ensure the tap works as expected
  * Ensure there are no issues related to running the tap, and loading the tap files
* Next exit running container, then run the command => `keg evf pack run --package evf:zen-614-tap-test --env prod`
  * This should run the same container in production
  * Ensure the tap works as expected
  * Ensure there are no issues related to running the tap, and loading the tap files
